agent-any/src/AnyAgentComplex.cc

```diff
@@ -432,7 +432,20 @@ AnyAgent::parseData (char const *&line, const YCPValue & syntax, bool optional)
     if ((line == 0) || (*line == 0))
 	line = getLine ();
     if (line == 0)
+    {
+        // handle special terms that do not need data like optional data
+        if (syntax->valuetype() == YT_TERM)
```

Comment:
right, I will reorder it. I think it is safe to just skip getLine if syntax is invalid, not?